File: VB/SaveDocument/CustomCommand.vb
Imports DevExpress.XtraRichEdit.Services
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.Commands
Imports System.Windows.Forms

Namespace SaveDocumentSample

'#Region "#savecommand"
    Public Class CustomRichEditCommandFactoryService
        Implements IRichEditCommandFactoryService

        Private ReadOnly service As IRichEditCommandFactoryService

        Private ReadOnly control As RichEditControl

        Public Sub New(ByVal control As RichEditControl, ByVal service As IRichEditCommandFactoryService)
            DevExpress.Utils.Guard.ArgumentNotNull(control, "control")
            DevExpress.Utils.Guard.ArgumentNotNull(service, "service")
            Me.control = control
            Me.service = service
        End Sub

        Public Function CreateCommand(ByVal id As RichEditCommandId) As RichEditCommand Implements IRichEditCommandFactoryService.CreateCommand
            If id = RichEditCommandId.FileSaveAs Then Return New CustomSaveDocumentAsCommand(control)
            Return service.CreateCommand(id)
        End Function
    End Class

    Public Class CustomSaveDocumentAsCommand
        Inherits SaveDocumentAsCommand

        Public Sub New(ByVal control As IRichEditControl)
            MyBase.New(control)
        End Sub

        Protected Overrides Sub ExecuteCore()
            Dim dialog As SaveFileDialog = New SaveFileDialog With {.Filter = "Rich Text Format Files (*.rtf)|*.rtf|All Files (*.*)|*.*", .FileName = "SavedDocument.rtf", .RestoreDirectory = True, .CheckFileExists = False, .CheckPathExists = True, .OverwritePrompt = True, .DereferenceLinks = True, .ValidateNames = True, .AddExtension = False, .FilterIndex = 1}
            dialog.InitialDirectory = "C:\Temp"
            If dialog.ShowDialog() = DialogResult.OK Then
                CType(Control, RichEditControl).SaveDocument(dialog.FileName, DocumentFormat.Rtf)
            End If
        'base.ExecuteCore();
        End Sub
    End Class
'#End Region  ' #savecommand
End Namespace

File: CS/SaveDocument/CustomCommand.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraRichEdit.Services;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.Commands;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace SaveDocumentSample
{
    #region #savecommand
    public class CustomRichEditCommandFactoryService : IRichEditCommandFactoryService
    {
        readonly IRichEditCommandFactoryService service;
        readonly RichEditControl control;
        public CustomRichEditCommandFactoryService(RichEditControl control, IRichEditCommandFactoryService service)
        {
            DevExpress.Utils.Guard.ArgumentNotNull(control, "control");
            DevExpress.Utils.Guard.ArgumentNotNull(service, "service");
            this.control = control;
            this.service = service;
        }
        public RichEditCommand CreateCommand(RichEditCommandId id)
        {
            if (id == RichEditCommandId.FileSaveAs)
                return new CustomSaveDocumentAsCommand(control);

            return service.CreateCommand(id);
        }
    }

    public class CustomSaveDocumentAsCommand : SaveDocumentAsCommand
    {
        public CustomSaveDocumentAsCommand(IRichEditControl control)
            : base(control) {}

        protected override void ExecuteCore()
        {
            using (XtraSaveFileDialog dialog = new XtraSaveFileDialog())
            {
                dialog.Filter = "Rich Text Format Files (*.rtf)|*.rtf|All Files (*.*)|*.*";
                dialog.FileName = "SavedDocument.rtf";
                dialog.RestoreDirectory = true;
                dialog.CheckFileExists = false;
                dialog.CheckPathExists = true;
                dialog.OverwritePrompt = true;
                dialog.DereferenceLinks = true;
                dialog.ValidateNames = true;
                dialog.AddExtension = false;
                dialog.FilterIndex = 1;
                dialog.InitialDirectory = "C:\\Temp";
                if (dialog.ShowDialog() == DialogResult.OK)
                {
                    ((RichEditControl)this.Control).SaveDocument(dialog.FileName, DocumentFormat.Rtf);
                }
            }
            //base.ExecuteCore();
        }
    }
        #endregion #savecommand
}
